src/app/filecoin-plus/allocators/schemas/AllocatorSchema.ts

```diff
@@ -42,8 +45,11 @@ export const AllocatorFileListMetaDataBaseSchema = z.array(
   AllocatorFileMetaDataBaseSchema,
 )
 
+export const AllocatorWithDatacapSchema = AllocatorSchema.merge(DatacapSchema)
```

Comment:
Nit: We usually use [extend](https://zod.dev/README?id=extend) for these operations.

Is there a reason for `merge` here?

Reply:
Ah, my bad! I thought they were interchangeable, but they have a slight difference.

Your first implementation was better than what I suggested, sorry!

```ts
A.merge(B)
// is equivalent to
A.extend(B.shape)
```

Reply:
Both seem to behave the same, though

> If the two schemas share keys, the properties of B overrides the property of A. The returned schema also inherits the "unknownKeys" policy (strip/strict/passthrough) and the catchall schema of B.